test/Concurrency/Runtime/async_task_cancellation_while_running.swift

```diff
@@ -0,0 +1,43 @@
+// RUN: %target-run-simple-swift(-Xfrontend -enable-experimental-concurrency  %import-libdispatch -parse-as-library)
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: libdispatch
+
+import Dispatch
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#endif
+
+func test_runDetached_cancel_while_child_running() async {
+  print(#function) // CHECK: test_runDetached_cancel_while_child_running
+  let h: Task.Handle<Bool> = Task.runDetached {
+    async let childCancelled: Bool = { () -> Bool in
+      sleep(3)
```

Comment:
not in love with sleeps in tests but I did not manage to build a "[CountdownLatch](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html)" using our Swift primitives so far... once I do I could replace all those sleeps with latches, help welcome :)

Reply:
Or I just write one using a semaphore... not sure that will work with all those suspends flying around...?